package/gluon-mesh-vpn-wireguard/luasrc/lib/gluon/upgrade/405-wireguard

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/lua
+local site = require 'gluon.site'
+local uci = require('simple-uci').cursor()
+local iputil = require 'gluon.iputil'
+local sysconfig = require 'gluon.sysconfig'
+
+uci:delete('wireguard', 'mesh_vpn', 'user')
+
+local add_groups
+
+local function generate_section_name(peer)
+  return "mesh_vpn_wg_" .. peer
+end
+
+local function generate_wg_iface_name(peer)
+  return "mesh-vpn-" .. peer
+end
+
+local function add_peer(group, name, config, count)
+  local ip = iputil.mac_to_ip("fe80::/64", sysconfig.primary_mac, 0x00, count)
+
+  uci:section('network', 'interface', generate_section_name(name) .. 'm', {
+    type = 'wireguard',
+    ifname = generate_wg_iface_name(count),
+    proto = 'gluon_mesh',
+  })
```

Comment:
Rather than adding all these interfaces to `/etc/config/network` here, they could be registered in netifd dynamically using ubus from wg post-up scripts or something similar.

Reply:
the whole concept hinges on ifup enabling the required interfaces periodically. For this they must be defined in /etc/config/network.

Reply:
Meh. It would be great if all interfaces could just be configured over ubus, and then ifup'd later. I'm not sure if netifd can do such a thing...

Reply:
I do not think it can. We'll have to keep them in the network config for now as discussed on 36c3